pkg/cmdlets/auth/login.go

```diff
@@ -226,22 +228,57 @@ func promptToken(credentials *mono_models.Credentials, out output.Outputer, prom
 func AuthenticateWithBrowser(out output.Outputer, auth *authentication.Auth, prompt prompt.Prompter) error {
 	logging.Debug("Authenticating with browser")
 
+	err := authenticateWithBrowser(out, auth, prompt, false)
+	if err != nil {
+		return errs.Wrap(err, "Error authenticating with browser")
+	}
+
+	out.Notice(locale.T("auth_device_success"))
+
+	return nil
+}
+
+// authenticateWithBrowser authenticates after signup if applicable.
+func authenticateWithBrowser(out output.Outputer, auth *authentication.Auth, prompt prompt.Prompter, signup bool) error {
 	response, err := model.RequestDeviceAuthorization()
 	if err != nil {
 		return locale.WrapError(err, "err_auth_device")
 	}
 
+	if response.VerificationURIComplete == nil {
+		return errs.New("Invalid response: Missing verification URL.")
+	}
+
+	verificationURL := *response.VerificationURIComplete
+	if signup {
+		// verificationURL is of the form:
+		//   https://platform.activestate.com/authorize/device?user-code=...
+		// Transform it to the form:
+		//   https://platform.activestate.com/create-account?nextRoute=%2Fauthorize%2Fdevice%3Fuser-code%3D...
+		parsedURL, err := url.Parse(verificationURL)
+		if err != nil {
+			return errs.Wrap(err, "Verification URL is not valid")
+		}
+
+		signupURL, err := url.Parse(constants.PlatformSignupURL)
+		if err != nil {
+			return errs.Wrap(err, "constants.PlatformSignupURL is not valid")
+		}
+		query := signupURL.Query()
+		query.Add("nextRoute", parsedURL.RequestURI())
+		signupURL.RawQuery = query.Encode()
+
+		verificationURL = signupURL.String()
+	}
```

Comment:
Would this be more well placed in a function of its own? One function for authenticating with the browser and another for signup.

Reply:
No, because they both use the same flow (request an auth token, call the browser, listen for response). The only difference between flows is the URL, hence the "if" you're commenting on. I didn't want to duplicate any code.